File: shamelagpt-android/app/src/test/java/com/shamelagpt/android/mock/MockScenarioMatrix.kt
package com.shamelagpt.android.mock

import com.shamelagpt.android.core.network.NetworkError

/**
 * Shared scenario matrix for Android unit/integration tests.
 */
enum class MockScenarioId(val wireId: String) {
    SUCCESS("success"),
    HTTP_400("http_400"),
    HTTP_401("http_401"),
    HTTP_403("http_403"),
    HTTP_404("http_404"),
    HTTP_429("http_429"),
    HTTP_500("http_500"),
    TIMEOUT("timeout"),
    OFFLINE("offline")
}

object MockScenarioMatrix {
    fun apply(scenario: MockScenarioId, repository: MockChatRepository) {
        repository.sendMessageResult = when (scenario) {
            MockScenarioId.SUCCESS -> Result.success(TestData.sampleChatResponse)
            MockScenarioId.HTTP_400 -> Result.failure(NetworkError.HttpError(400))
            MockScenarioId.HTTP_401 -> Result.failure(NetworkError.HttpError(401))
            MockScenarioId.HTTP_403 -> Result.failure(NetworkError.HttpError(403))
            MockScenarioId.HTTP_404 -> Result.failure(NetworkError.HttpError(404))
            MockScenarioId.HTTP_429 -> Result.failure(NetworkError.HttpError(429))
            MockScenarioId.HTTP_500 -> Result.failure(NetworkError.HttpError(500))
            MockScenarioId.TIMEOUT -> Result.failure(NetworkError.Timeout)
            MockScenarioId.OFFLINE -> Result.failure(NetworkError.NoConnection)
        }
    }
}

File: shamelagpt-ios/shamelagptTests/Mocks/MockScenarioMatrix.swift
import Foundation
@testable import ShamelaGPT

enum MockScenarioID: String, CaseIterable {
    case success = "success"
    case http400 = "http_400"
    case http401 = "http_401"
    case http403 = "http_403"
    case http404 = "http_404"
    case http429 = "http_429"
    case http500 = "http_500"
    case timeout = "timeout"
    case offline = "offline"
}

/// Shared scenario matrix for iOS unit/integration tests.
enum MockScenarioMatrix {
    static func apply(_ scenario: MockScenarioID, to api: MockAPIClient) {
        api.shouldFail = false
        api.errorToThrow = NetworkError.noConnection

        switch scenario {
        case .success:
            api.shouldFail = false
        case .http400:
            fail(api, with: .httpError(statusCode: 400))
        case .http401:
            fail(api, with: .httpError(statusCode: 401))
        case .http403:
            fail(api, with: .httpError(statusCode: 403))
        case .http404:
            fail(api, with: .httpError(statusCode: 404))
        case .http429:
            fail(api, with: .httpError(statusCode: 429))
        case .http500:
            fail(api, with: .httpError(statusCode: 500))
        case .timeout:
            fail(api, with: .timeout)
        case .offline:
            fail(api, with: .noConnection)
        }
    }

    private static func fail(_ api: MockAPIClient, with error: NetworkError) {
        api.shouldFail = true
        api.errorToThrow = error
    }
}
